src/java.base/share/classes/com/sun/crypto/provider/DHPrivateKey.java

```diff
@@ -106,8 +106,11 @@
         this.g = g;
         this.l = l;
         try {
-            this.key = new DerValue(DerValue.tag_Integer,
-                                    this.x.toByteArray()).toByteArray();
+            byte[] xbytes = x.toByteArray();
+            DerValue val = new DerValue(DerValue.tag_Integer, xbytes);
+            this.key = val.toByteArray();
+            val.clear();
+            Arrays.fill(xbytes, (byte)0);
             encode();
         } catch (IOException e) {
             throw new ProviderException("Cannot produce ASN.1 encoding", e);
```

Comment:
Supposedly the IOException should never happen? Otherwise the Arrays.fill(...) call may not happen.

Reply:
No, it should never happen. `DerValue::toByteArray` claims it might throw an IOE because `DerOutputStream::putLength` claims so, but you can see the latter never throws one. I thought about removing those `throws IOE` but there are too many. We can do it in another code change.

Reply:
Ok.